include/ruby/internal/core/robject.h

```diff
@@ -124,14 +135,30 @@ struct RObject {
             struct st_table *iv_index_tbl;
         } heap;
 
+#if USE_RVARGC
+        VALUE ary[1];
```

Comment:
Can you add comments about this?  I don't understand what this field is for and why it's only `1`.